Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,376 @@
+# coding=utf-8
+"""
+Module of Credit mining function testing
+
+Written by Ardhi Putra Pratama H
+"""
+import binascii
+import os
+import shutil
+import time
+from hashlib import sha1
+
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.Utilities.network_utils import get_random_port
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST, NTFY_VOTECAST
+from Tribler.Main.Utility.GuiDBTuples import CollectedTorrent
+from Tribler.Policies.BoostingManager import BoostingManager, BoostingSettings
+from Tribler.Test.test_as_server import TestAsServer, TESTS_DATA_DIR
+from Tribler.community.allchannel.community import AllChannelCommunity
+from Tribler.dispersy.dispersy import Dispersy
+from Tribler.dispersy.endpoint import ManualEnpoint
+from Tribler.dispersy.member import DummyMember
+from Tribler.dispersy.util import blocking_call_on_reactor_thread
+
+FILE_DIR = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+TORRENT_FILE = os.path.join(TESTS_DATA_DIR, "ubuntu-15.04-desktop-amd64.iso.torrent")
```

Comment:
It is good to refer to other tests if it helps avoiding duplicated code and maintainability.  If a torrent needs to be replaced, it's way better to just change one constant instead of having to fish for references all over the place.